agent/config/parse_linux.go

```diff
@@ -22,3 +22,8 @@ func parseGMSACapability() bool {
 func parseFSxWindowsFileServerCapability() bool {
 	return false
 }
+
+// GetOperatingSystemFamily() returns "linux" as operating system family for linux based ecs instances
+func GetOperatingSystemFamily() string {
```

Comment:
why do you need this new function? you can access the `OSType` const directly

Reply:
As per conversation, the runtime fails without this. Its needed